### PR TITLE
improve main

### PR DESCRIPTION
**Pull request by zhangtianli2006**

### CODE REVIEW

File: include/io/fs/main-path.h [hunk 1]
```diff
@@ -24,14 +24,13 @@ namespace frontend {
 
 class CurrentPath {
 public:
+    CurrentPath();
+    ~CurrentPath();
```
Comment (szdytom) [inline include/io/fs/main-path.h:28]: for what?

File: src/main.cc [hunk 2]
```diff
@@ -15,17 +15,23 @@
...
 #include <gflags/gflags.h>
 #include <unistd.h>
+
 #include "io/io.h"
 #include "slide/slide.h"
 
-DEFINE_string(input, "/dev/stdin", "Input markdown file.");
-DEFINE_string(output, "/dev/stdout", "Output directory. Where to place render slide");
+DEFINE_string(input, "", "Inpput markdown file.");
+DEFINE_string(output, "", "Output directory. Where to place rendered slide");
```
Comment (szdytom) [inline src/main.cc:31]: you can't read from "", can you?
```suggestion
DEFINE_string(input, "./", "Input markdown file.");
DEFINE_string(output, "./", "Output directory. Where to place rendered slide");
```

**Review by szdytom — CHANGES_REQUESTED** [review]

File: src/main.cc [hunk 3]
```diff
@@ -33,6 +33,59 @@ DEFINE_bool(license, false, "Output program license");
 DEFINE_bool(config, false, "Show config");
 DEFINE_bool(path, false, "Show current path");
 
+void printIcon(bool large) {
+    if (large) {
+        std::string icon = "\n"
+                           "    +-------------------+  │\n"
+                           "+-------------------+###|  │  Lightening Slide V0.1a\n"
```
Comment (szdytom) [inline src/main.cc:40]: It isn't a good idea to write version text directly inside the program, I think.
Well, there is something like `GFLAGS_NAMESPACE::SetVersionText`

Comment (zhangtianli2006) [reply src/main.cc:40]: What's that

Comment (zhangtianli2006) [reply src/main.cc:40]: Maybe you should explain how `GFLAGS_NAMESPACE::SetVersionText` so that I can fix this.

File: include/io/fs/fs.h [hunk 4]
```diff
@@ -24,7 +24,7 @@ namespace frontend {
 
 bool isFileExist(const std::string& filepath);
 void moveFile(const std::string& originFilename, const std::string& newFilename);
-void copyFile(const std::string& originFilename, const std::string& newFilename);
+void copyFile(const std::string& originFilename, const std::string& newFilename, const bool withDir);
```
Comment (szdytom) [inline include/io/fs/fs.h:27]: unexcepted const.
```suggestion
void copyFile(const std::string& originFilename, const std::string& newFilename, bool withDir);
```

or maybe
```suggestion
void copyFile(const std::string& originFilename, const std::string& newFilename, const bool& withDir);
```

File: src/io/fs/fs.cc [hunk 5]
```diff
@@ -37,8 +37,13 @@ void frontend::moveFile(const std::string& originFilename, const std::string& ne
...
 
-void frontend::copyFile(const std::string& originFilename, const std::string& newFilename) {
-    std::string command(fmt::format("cp -r {} {}", originFilename, newFilename));
+void frontend::copyFile(const std::string& originFilename, const std::string& newFilename, const bool withDir) {
+    std::string command;
+    if (withDir) {
+        command = fmt::format("cp -r {} {}", originFilename, newFilename);
+    } else {
+        command = fmt::format("cp {} {}", originFilename, newFilename);
+    }
```
Comment (szdytom) [inline src/io/fs/fs.cc:46]: I wonder for what reason.

File: src/main.cc [hunk 6]
```diff
@@ -109,6 +111,11 @@ int main(int argc, char** argv) {
         return 0;
     }
 
+    if (FLAGS_install.size() > 0) {
+        frontend::copyFile(FLAGS_install, fmt::format("{}extension", frontend::CurrentPath::get()), false);
```
Comment (szdytom) [inline src/main.cc:115]: You should copy costum ext to `./data/extension`, not defualt extension

File: src/io/fs/fs.cc (hunk 5 above)
Comment (zhangtianli2006) [reply src/io/fs/fs.cc:46]: Not always we want to copy with dir.
`frontend::copyFile(FLAGS_install, fmt::format("{}extension", frontend::CurrentPath::get()), false);`
copying the hole dir is not allowed or useless here.

File: src/main.cc (hunk 3 above)
Comment (zhangtianli2006) [reply src/main.cc]: solved

**Comment by zhangtianli2006** [comment]
I'm adding a build shell to /script, if anyone see this, please don't approve.

File: src/main.cc [hunk 7]
```diff
@@ -15,33 +15,155 @@
...
+                fmt::print(bg(fmt::color::light_blue) | fg(fmt::color::sky_blue) | fmt::emphasis::bold, "{}", icon[i]);
+            } else {
+                fmt::print("{}", icon[i]);
+            }
+        }
+    }
+}
 
 int main(int argc, char** argv) {
+    GFLAGS_NAMESPACE::SetVersionString("V0.1a");
```
Comment (gzezFISHER) [inline src/main.cc:91]:
```suggestion
    GFLAGS_NAMESPACE::SetVersionString("V0.1b");
```

File: src/main.cc [hunk 8]
```diff
@@ -15,33 +15,155 @@
...
+    }
+
+    if (FLAGS_path) {
+        fmt::print(fg(fmt::color::light_sky_blue) | fmt::emphasis::bold, "LighteningSlide is at:\n");
+        fmt::print(fmt::emphasis::underline, "{}\n", frontend::CurrentPath::get());
+        return 0;
+    }
+
+    if (FLAGS_install.size() > 0) {
+        frontend::copyFile(FLAGS_install, fmt::format("{}extension", frontend::CurrentPath::get()), false);
```
Comment (gzezFISHER) [inline src/main.cc:122]:
```suggestion
        frontend::copyFile(FLAGS_install, fmt::format("{}data/extension", frontend::CurrentPath::get()), false);
```

**Review by gzezFISHER — APPROVED** [review]

**Review by szdytom — APPROVED** [review]

**Pull request merged**